rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/FileDownloader.java

```diff
@@ -94,9 +100,15 @@ private void initializeMetrics()
       log.info("Processing Request for File : " + path.toString() + " LocalFile : " + localPath);
       ByteBuffer directWriteBuffer = bufferPool.getBuffer(diskReadBufferSize);
 
-      FileDownloadRequestChain requestChain = new FileDownloadRequestChain(bookKeeper, fs, localPath,
-          directWriteBuffer, conf, context.getRemoteFilePath(), context.getFileSize(),
-          context.getLastModifiedTime());
+      BookKeeperService.Client bookKeeperClient = null;
```

Comment:
This should be a `RetryingBookkeeperClient` instead, which wraps client calls with retry logic.

Reply:
I'll make the change